### ISSUE TITLE
Subclass always invokes superclass private constructor

### ISSUE THREAD

**Issue by ldcouto** [overturetool/overture#581]
If you have the following model:

```
class A
operations
private A : () ==> A
A() == error;

public A : int ==> A
A (x) == skip;
end A

class B is subclass of A
operations
public B : int ==>B
B(x) == A`(x)
end B
```

When you try to run `new B(0)`, you get a  `Error 4163: Cannot inherit private constructor in 'A' (C:\Users\CoutoLD\dev\workspaces\ovt238\Test\A.vdmpp) at line 4:9`. If you make it public, the error statement is hit.

I find this odd. Why is the empty constructor being invoked when I explicitly tell it to invoke the one with parameters?

I have a model where I want to disable the empty constructor for a class but I still want to subclass it. This seems impossible.

**Comment by nickbattle** [comment]
I agree this is strange and not what you would want or expect, but similar behaviour has been noted in VDMTools. Have a look at the (closed) bug #1. These VDM++ OO bugs were all closed in anticipation of some sensible OO semantics work,

**Comment by ldcouto** [comment]
Right. It doesn't prevent me from doing anything so it's not a huge deal.

I support closing this issue. For consistency :)
